samples/dnn/action_recognition.py

```diff
@@ -0,0 +1,280 @@
+import os
+import copy
+import shutil
+import numpy as np
+import cv2 as cv
+import argparse
+
+parser = argparse.ArgumentParser(add_help=False)
+parser.add_argument('--input', help='Path to input video file.')
+parser.add_argument('--model', help='Path to model.')
+parser.add_argument('--classes', help='Path to classes list.')
+parser.add_argument('--temporal_unit', default=5, help='Averages the scores over temporal_unit x sample_duration clips.')
+parser.add_argument('--sample_duration', default=16, help='Number of frames for action detection.')
+
+# To get net download original repository https://github.com/kenshohara/video-classification-3d-cnn-pytorch.git
+# For correct ONNX export modify file: video-classification-3d-cnn-pytorch/models/resnet.py 
+# change
+# - def downsample_basic_block(x, planes, stride):
+# -     out = F.avg_pool3d(x, kernel_size=1, stride=stride)
+# -     zero_pads = torch.Tensor(out.size(0), planes - out.size(1),
+# -                              out.size(2), out.size(3),
+# -                              out.size(4)).zero_()
+# -     if isinstance(out.data, torch.cuda.FloatTensor):
+# -         zero_pads = zero_pads.cuda()
+# - 
+# -     out = Variable(torch.cat([out.data, zero_pads], dim=1))
+# -     return out
+
+# To 
+# + def downsample_basic_block(x, planes, stride):
+# +     out = F.avg_pool3d(x, kernel_size=1, stride=stride)
+# +     out = F.pad(out, (0, 0, 0, 0, 0, 0, 0, int(planes - out.size(1)), 0, 0), "constant", 0)
+# +     return out
+
+# To ONNX export use torch.onnx.export(model, inputs, model_name) 
+
+
+def parse_video(video_path):
+    cap = cv.VideoCapture(video_path)
+    fps = cap.get(cv.CAP_PROP_FPS)
+    if os.path.exists('tmp'):
+        shutil.rmtree('tmp')
+    os.mkdir('tmp')
+    i = 0
+    while cv.waitKey(1) < 0:
+        hasFrame, frame = cap.read()
+        if hasFrame:
+            cv.imwrite('tmp/image_{:05}.jpg'.format(i), frame)
+            i += 1
+            if cv.waitKey(1) & 0xFF == ord('q'):
+                break
+        else:
+            break
+    return fps
```

Comment:
Please estimate an efficiency of model so we can design a sample in a different way. It should be as minimal as possible so we need to render video but not save it.